test/integration/helpers.bash

```diff
@@ -88,7 +88,7 @@ function swarm_join() {
 		SWARM_JOIN_PID[$i]=$!
 		((++i))
 	done
-	wait_until_swarm_joined $i
+	wait_until_swarm_joined $i "$@"
```

Comment:
I'm wondering if `wait_until_swarm_joined` could use the new `retry` function (just like `wait_until_reachable`)

Maybe something like ```retry 10 1 [ `swarm list "$@" | wc -l` -eq $i ]```